File: 0078_Subsets/SubsetsBFS.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        var ans = new List<IList<int>>();
        ans.Add(new List<int>());
        
        foreach(var v in nums){
            int count = ans.Count;
            for(int i=0;i<count;i++){
                var subSet = new List<int>(ans[i]);
                subSet.Add(v);
                ans.Add(subSet);
            }
        }
        
        return ans;
    }
}

File: 0078_Subsets/SubsetsBFS.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>());
        for(int n : nums){
            int size = ans.size();
            for(int i=0; i<size; i++){
                List<Integer> subset = new ArrayList<>(ans.get(i));
                subset.add(n);
                ans.add(subset);
            }
        }
        return ans;
    }
}